internal/pipe/snapcraft/snapcraft.go

```diff
@@ -38,16 +38,21 @@ var ErrNoSummary = errors.New("no summary provided for snapcraft")
 // Metadata to generate the snap package.
 type Metadata struct {
 	Name          string
+	Title         string
 	Version       string
 	Summary       string
 	Description   string
+	Icon          string
 	Base          string `yaml:",omitempty"`
 	License       string `yaml:",omitempty"`
 	Grade         string `yaml:",omitempty"`
 	Confinement   string `yaml:",omitempty"`
 	Architectures []string
+	Assumes       []string                  `yaml:",omitempty"`
 	Layout        map[string]LayoutMetadata `yaml:",omitempty"`
 	Apps          map[string]AppMetadata
+	Environment   map[string]interface{} `yaml:",omitempty"`
```

Comment:
I think this needs be in the AppMetadata struct...

Reply:
Ah indeed! Weirdly this did actually work but I guess that's an undocumented feature that should not be relied upon.

I'm currently on holiday but I'll update this PR when I'm back in 10 days.

Reply:
Just FYI it is in the schema: https://github.com/snapcore/snapcraft/blob/eb0bf1276cf2ac22ee8db5dc27dfadc0d0433cdd/schema/snapcraft.json#L490